### ISSUE TITLE
chrome.instanceID not responding

### ISSUE THREAD

**Issue by jasonsavard** [brave/brave-browser#3952]
## Description 
All chrome.instanceID.* methods are not responding to callbacks or runtime.lastError

## Steps to Reproduce
chrome.instanceID.getID(function (id) {
    // never enters here
});

## Expected result:
Should execute the callback and return id or runtime.lastError

## Reproduces how often: 
Everytime

## Brave version (brave://version info)
Latest

**Comment by rebron** [comment]
@jasonsavard Can you give us more info? Is this for extension development? Can you attach what you're trying to do?

**Comment by jasonsavard** [comment]
Yes it's for extensions, here is workaround that I must use for Brave browsers...

`function getInstanceId(storage) {
    return new Promise(async (resolve, reject) => {
        const instanceId = await storage.get("instanceId");
        if (instanceId) {
            resolve(instanceId);
        } else {
            chrome.instanceID.getID(instanceId => {
                if (chrome.runtime.lastError) {
                    const error = new Error("Problem getting instanceid: " + chrome.runtime.lastError.message);
                    console.error(error);
                    reject(error)
                } else {
                    clearTimeout(window.instanceIdTimeout);
                    resolve(instanceId);
                }
            });
    
            // seems Brave browser doesn't respond to success or failure
            window.instanceIdTimeout = setTimeout(() => {
                reject("instanceId not responding");
            }, seconds(2));
        }
    }).catch(error => {
        console.warn("Generating instanceId");
        const instanceId = getUUID();
        storage.set("instanceId", instanceId);
        return instanceId;
    });
}`

**Comment by Laruxo** [comment]
I can confirm that I have the same issue.
Since `chrome.instanceID` requires `gcm` permissions, I guess this is related to https://github.com/brave/brave-browser/issues/1716

**Comment by bsclifton** [comment]
This issue hasn't had an update in a while - it's met the criteria for being stale. The issue is going to be closed for now. If this is still a problem, please leave a comment or re-open the issue. Thanks!
